Cabal/Distribution/PackageDescription.hs

```diff
@@ -631,15 +619,6 @@ emptyTestSuite = mempty
 hasTests :: PackageDescription -> Bool
 hasTests = any (buildable . testBuildInfo) . testSuites
 
--- | Get all the enabled test suites from a package.
-enabledTests :: PackageDescription -> [TestSuite]
-enabledTests = filter testEnabled . testSuites
-
--- | Perform an action on each buildable 'TestSuite' in a package.
-withTest :: PackageDescription -> (TestSuite -> IO ()) -> IO ()
```

Comment:
Correct. Unfortunately a backwards compatible interface is not possible because you really do need the CLBI to determine what is enabled.